docs/docs/development/new-behavior.md

```diff
@@ -320,38 +317,47 @@ The fifth cell of `BEHAVIOR_DT_INST_DEFINE` can be set to `NULL` instead if inst
+Most behavior drivers are invoked according to the central half's [locality](#api-structure), and are therefore stored after the line `if ((NOT CONFIG_ZMK_SPLIT) OR CONFIG_ZMK_SPLIT_ROLE_CENTRAL)` in the form, `target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_<BEHAVIOR_NAME> app PRIVATE src/behaviors/<behavior_name>.c)`, as shown below.
 
 ```txt title="app/CmakeLists.txt"
 if ((NOT CONFIG_ZMK_SPLIT) OR CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
-  target_sources(app PRIVATE src/behaviors/behavior_key_press.c)
-  target_sources(app PRIVATE src/behaviors/behavior_hold_tap.c)
-  target_sources(app PRIVATE src/behaviors/behavior_sticky_key.c)
-  target_sources(app PRIVATE src/behaviors/behavior_caps_word.c)
-  target_sources(app PRIVATE src/behaviors/behavior_key_repeat.c)
-  target_sources(app PRIVATE src/behaviors/behavior_momentary_layer.c)
-  target_sources(app PRIVATE src/behaviors/behavior_mod_morph.c)
-  target_sources(app PRIVATE src/behaviors/behavior_outputs.c)
-  target_sources(app PRIVATE src/behaviors/behavior_tap_dance.c)
-  target_sources(app PRIVATE src/behaviors/behavior_toggle_layer.c)
-  target_sources(app PRIVATE src/behaviors/behavior_to_layer.c)
-  target_sources(app PRIVATE src/behaviors/behavior_transparent.c)
-  target_sources(app PRIVATE src/behaviors/behavior_none.c)
-  target_sources(app PRIVATE src/behaviors/behavior_sensor_rotate_key_press.c)
-  target_sources(app PRIVATE src/combo.c)
-  target_sources(app PRIVATE src/conditional_layer.c)
-  target_sources(app PRIVATE src/keymap.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_KEY_PRESS app PRIVATE src/behaviors/behavior_key_press.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_KEY_TOGGLE app PRIVATE src/behaviors/behavior_key_toggle.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_HOLD_TAP app PRIVATE src/behaviors/behavior_hold_tap.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_STICKY_KEY app PRIVATE src/behaviors/behavior_sticky_key.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_CAPS_WORD app PRIVATE src/behaviors/behavior_caps_word.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_KEY_REPEAT app PRIVATE src/behaviors/behavior_key_repeat.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_MACRO app PRIVATE src/behaviors/behavior_macro.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_MOMENTARY_LAYER app PRIVATE src/behaviors/behavior_momentary_layer.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_MOD_MORPH app PRIVATE src/behaviors/behavior_mod_morph.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_OUTPUTS app PRIVATE src/behaviors/behavior_outputs.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_TOGGLE_LAYER app PRIVATE src/behaviors/behavior_toggle_layer.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_TO_LAYER app PRIVATE src/behaviors/behavior_to_layer.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_TRANSPARENT app PRIVATE src/behaviors/behavior_transparent.c)
+  target_sources_ifdef(CONFIG_ZMK_BEHAVIOR_NONE app PRIVATE src/behaviors/behavior_none.c)
```

Comment:
This is not quite matching https://github.com/zmkfirmware/zmk/blob/cb427b8a5c99bb79dfaff6be1673142b73bc0a7b/app/CMakeLists.txt, but it is a moving target anyway. How about a `/* ... */` in the beginning and end, and reduce the specific examples to a few?

Reply:
That's a good idea. I'll update it.